pymc3/distributions/continuous.py

```diff
@@ -738,40 +729,31 @@ def logp(self, value):
         -------
         TensorVariable
         """
-        mu = self.mu
-        sigma = self.sigma
-
-        norm = self._normalization()
-        logp = Normal.dist(mu=mu, sigma=sigma).logp(value) - norm
-
-        bounds = [sigma > 0]
-        if self.lower_check is not None:
-            bounds.append(value >= self.lower)
-        if self.upper_check is not None:
-            bounds.append(value <= self.upper)
-        return bound(logp, *bounds)
-
-    def _normalization(self):
-        mu, sigma = self.mu, self.sigma
-
-        if self.lower_check is None and self.upper_check is None:
-            return 0.0
-
-        if self.lower_check is not None and self.upper_check is not None:
-            lcdf_a = normal_lcdf(mu, sigma, self.lower)
-            lcdf_b = normal_lcdf(mu, sigma, self.upper)
-            lsf_a = normal_lccdf(mu, sigma, self.lower)
-            lsf_b = normal_lccdf(mu, sigma, self.upper)
-
-            return at.switch(self.lower > 0, logdiffexp(lsf_a, lsf_b), logdiffexp(lcdf_b, lcdf_a))
-
-        if self.lower_check is not None:
-            return normal_lccdf(mu, sigma, self.lower)
+        unbounded_lower = isinstance(lower, TensorConstant) and np.all(lower.value == -np.inf)
+        unbounded_upper = isinstance(upper, TensorConstant) and np.all(upper.value == np.inf)
+
+        if not unbounded_lower and not unbounded_upper:
+            lcdf_a = normal_lcdf(mu, sigma, lower)
+            lcdf_b = normal_lcdf(mu, sigma, upper)
+            lsf_a = normal_lccdf(mu, sigma, lower)
+            lsf_b = normal_lccdf(mu, sigma, upper)
+            norm = at.switch(lower > 0, logdiffexp(lsf_a, lsf_b), logdiffexp(lcdf_b, lcdf_a))
+        elif not unbounded_lower:
+            norm = normal_lccdf(mu, sigma, lower)
+        elif not unbounded_upper:
+            norm = normal_lcdf(mu, sigma, upper)
         else:
-            return normal_lcdf(mu, sigma, self.upper)
-
-    def _distr_parameters_for_repr(self):
-        return ["mu", "sigma", "lower", "upper"]
+            norm = 0.0
+
+        logp = Normal.logp(value, mu=mu, sigma=sigma) - norm
+        bounds = []
+        if not unbounded_lower:
+            bounds.append(value >= lower)
+        if not unbounded_upper:
+            bounds.append(value <= upper)
+        if not unbounded_lower and not unbounded_upper:
+            bounds.append(lower <= upper)
```

Comment:
This logic can be placed in the `if else` branches above, no point in double checking